website/content/maintainers/security/security-guidelines.md

```diff
@@ -0,0 +1,241 @@
+
+The foundation of a project is its source code and it is essential to ensure the integrity of the source code. Source code repositories such as GitHub allow contributions from numerous members from all across the world in a single repository which is a boon in itself, however if not leveraged securely, it can be a bane. The key to doing so is protecting the repository where the source code lies and introducing changes to the repository in a controlled and secure manner.
+
+This section outlines several measures that can be taken to ensure authorized members have access to the code repository, changes are suggested, suggested changes are reviewed, and changes are introduced to the repository in a secure and controlled manner.
+
+It should be noted that secure device management is assumed - if you have a remote access trojan on your laptop, many of these countermeasures are subverted.
+
+## 1.1 Access management
+
+### Enable Role Based Access Control (RBAC)
+
+Define roles and associated access controls based upon the different personas interacting with the code repositories. The roles should be assigned following least privilege on a need-to know basis based on their assigned responsibilities. For example, GitHub provides roles such as Owner, Maintainer, Developer, Reviewer, Approver, and Guest. Each role should then be given fine-grained permissions with regards to repository access control.
+
+### Strong Authentication mechanisms
+
+Strong authentication mechanisms are key to ensure accounts are not susceptible to several attacks including but not limited to account takeover. We highly recommend the use of password-protected SSH keys or a personal access token (PAT).
+
+However, for certain non-code intensive projects which specifically require accommodation that cannot be made possible using recommended methods (such as access to Github.com via browser), we recommend the use of unique and complex passwords (complexity in terms of a combination of alphanumeric, special characters as well as length). For details regarding configuring an authentication mechanism for GitHub, please refer to GitHub's documentation [GitHub Docs - About authentication to GitHub](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-authentication-to-github).
+
+### Two Factor Authentication/Multi-factor Authentication
+
+Introducing an additional factor such as "something you have" or "somewhere you are" in the authentication process proves to be of higher effectiveness than relying on one factor "something you know" such as passwords, hence we urge configuring two factor or multi-factor authentication for your accounts, especially for any privileged accounts. For details regarding configuring a 2FA for GitHub, please refer to GitHub's documentation [GitHub Docs - Securing your account with two-factor authentication (2FA)](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa).
```

Comment:
GitHub is beginning to enforce this. It might make sense to reflect that this isn't an urging, but impending requirement by GitHub for any CNCF (or large open source) project. 

See https://github.blog/2023-03-09-raising-the-bar-for-software-security-github-2fa-begins-march-13/ for more info

I do think it's okay to leave as-is, but within a few months the wording should change.